cmd/erasure-healing-common.go

```diff
@@ -347,13 +347,6 @@ func disksWithAllParts(ctx context.Context, onlineDisks []StorageAPI, partsMetad
 					dataErrs[i] = errFileCorrupt
 					continue
 				}
-
-				// Since erasure.Distribution is trustable we can fix the mismatching erasure.Index
```

Comment:
@harshavardhana is this intentional ?

Reply:
yes @vadmeste this situation cannot be fixed via healing, The issue is similar to EC:0 to EC:4 - you need to use existing shards to rewrite the incorrect positions.

This is not possible to do in any manner while reading a shard and overwriting the same shard.